File: llm_generated_code/ChatGPT/CPP/Level2/Q3.cpp
#include <iostream>
#include <vector>
using namespace std;

// Function to check if a given digit is large (i.e., between 5 and 9 inclusive)
bool is_large_digit(int digit) {
    return digit >= 5 && digit <= 9;
}

// Function to check if a number can be split into two large numbers with the same number of digits
bool can_be_sum_of_large_numbers(int x) {
    string x_str = to_string(x);
    int n = x_str.length();

    // Check if x can be represented as the sum of two large integers
    for (int len_x = 1; len_x <= n; ++len_x) {
        bool possible = true;
        int carry = 0;

        for (int i = len_x - 1; i >= 0; --i) {
            int digit_x = (i < x_str.length() ? x_str[i] - '0' : 0) - carry;
            carry = 0;

            if (digit_x < 10 && is_large_digit(5 + digit_x % 10)) {
                continue;
            }
            possible = false;
            break;
        }

        if (possible) {
            return true;
        }
    }

    return false;
}

int main() {
    int t;
    cin >> t;
    vector<string> results;

    for (int i = 0; i < t; ++i) {
        int x;
        cin >> x;
        if (can_be_sum_of_large_numbers(x)) {
            results.push_back("Yes");
        } else {
            results.push_back("No");
        }
    }

    for (const auto& result : results) {
        cout << result << endl;
    }

    return 0;
}

File: llm_generated_code/ChatGPT/Python/Level2/Q3.py
def can_be_sum_of_large_numbers(x):
    str_x = str(x)
    length = len(str_x)
    
    # If x is too small to be a sum of two large numbers
    if length == 1:
        return "No"
    
    # Iterate through each possible length
    for d in range(1, length + 1):
        if d > 1:
            # Split x into two numbers, each with d digits
            num1, num2 = x // (10 ** (d - 1)), x % (10 ** (d - 1))
            
            # Check if both numbers have only large digits (5 to 9)
            if all(5 <= int(digit) <= 9 for digit in str(num1)) and all(5 <= int(digit) <= 9 for digit in str(num2)):
                return "Yes"
    
    return "No"

# Main function to handle multiple test cases
def main():
    t = int(input().strip())
    results = []
    
    for _ in range(t):
        x = int(input().strip())
        results.append(can_be_sum_of_large_numbers(x))
    
    for result in results:
        print(result)

# Call the main function
if __name__ == "__main__":
    main()
